clients/src/main/java/org/apache/kafka/common/record/MemoryRecords.java

```diff
@@ -162,79 +163,82 @@ private static FilterResult filterTo(TopicPartition partition, Iterable<MutableR
-            // recopy the messages to the destination buffer.
-            byte batchMagic = batch.magic();
-            List<Record> retainedRecords = new ArrayList<>();
-
-            final BatchFilterResult iterationResult = filterBatch(batch, decompressionBufferSupplier, filterResult, filter,
-                    batchMagic, true, retainedRecords);
-            boolean containsTombstones = iterationResult.containsTombstones;
-            boolean writeOriginalBatch = iterationResult.writeOriginalBatch;
-            long maxOffset = iterationResult.maxOffset;
-
-            if (!retainedRecords.isEmpty()) {
-                // we check if the delete horizon should be set to a new value
-                // in which case, we need to reset the base timestamp and overwrite the timestamp deltas
-                // if the batch does not contain tombstones, then we don't need to overwrite batch
-                boolean needToSetDeleteHorizon = batch.magic() >= RecordBatch.MAGIC_VALUE_V2 && (containsTombstones || containsMarkerForEmptyTxn)
-                    && !batch.deleteHorizonMs().isPresent();
-                if (writeOriginalBatch && !needToSetDeleteHorizon) {
-                    batch.writeTo(bufferOutputStream);
-                    filterResult.updateRetainedBatchMetadata(batch, retainedRecords.size(), false);
-                } else {
-                    final MemoryRecordsBuilder builder;
-                    long deleteHorizonMs;
-                    if (needToSetDeleteHorizon)
-                        deleteHorizonMs = filter.currentTime + filter.deleteRetentionMs;
-                    else
-                        deleteHorizonMs = batch.deleteHorizonMs().orElse(RecordBatch.NO_TIMESTAMP);
-                    builder = buildRetainedRecordsInto(batch, retainedRecords, bufferOutputStream, deleteHorizonMs);
-
-                    MemoryRecords records = builder.build();
-                    int filteredBatchSize = records.sizeInBytes();
-                    if (filteredBatchSize > batch.sizeInBytes() && filteredBatchSize > maxRecordBatchSize)
-                        log.warn("Record batch from {} with last offset {} exceeded max record batch size {} after cleaning " +
-                                        "(new size is {}). Consumers with version earlier than 0.10.1.0 may need to " +
-                                        "increase their fetch sizes.",
-                                partition, batch.lastOffset(), maxRecordBatchSize, filteredBatchSize);
-
-                    MemoryRecordsBuilder.RecordsInfo info = builder.info();
-                    filterResult.updateRetainedBatchMetadata(info.maxTimestamp, info.shallowOffsetOfMaxTimestamp,
-                            maxOffset, retainedRecords.size(), filteredBatchSize);
+        try (ByteBufferOutputStream bufferOutputStream = new ByteBufferOutputStream(destinationBuffer)) {
```

Comment:
Note that there is no logic change here.

We only added the ByteBufferOutputStream into a try-with-resources block and everything got indented by a tab inside that block.